File: kotlin/pig-latin/src/main/kotlin/PigLatin.kt
package PigLatin

val re1 = Regex("""^([aeiou]|xr|yt)""")
val re2 = Regex("""^([^aeiou]+(?=y)|[^aeiou]*qu|[^aeiou]+)([a-z]+)""")

fun translate(phrase: String): String {
  var res = arrayListOf<String>()

  for (word in phrase.split(" ")) {
    if (re1.containsMatchIn(word)) {
      res.add(word + "ay")
    } else {
      res.add(re2.replace(word, "$2$1") + "ay")
    }
  }
  return res.joinToString(" ")
};

File: typescript/pig-latin/pig-latin.ts
export default class PigLatin {
  static translate = (phrase: string) =>
    phrase.split(' ').map((word) => /^([aeiou]|xr|yt)/.test(word) ? word + 'ay' :
      word.replace(/^([^aeiou]+(?=y)|[^aeiou]*qu|[^aeiou]+)([a-z]+)/, '$2$1') + 'ay'
    ).join(' ')
}
